doc/RELEASE_WALKTHROUGH.rst.txt

```diff
@@ -0,0 +1,170 @@
+This file contains a walkthrough of the NumPy 1.12.0 release on Fedora Linux.
+The commands can be copied into the command line, but be sure to
+replace 1.12.0 by the correct version.
+
+Release  Walkthrough
+====================
+
+Building the release
+--------------------
+
+Checkout the branch for the release, make sure it is up to date, and clean the
+repository::
+
+    $ git checkout maintenance/1.12.x
+    $ git pull upstream maintenance/1.12.x
+    $ git submodule update
+    $ git clean -xdf
+
+Look at the git log to get the hash of the last commit in the release, then
+check it out::
+
+    $ git log
+    $ git checkout f59a2fbbd497dbaf3a2e405a41e52dc5e331bfa7
+
+Edit pavement.py and setup.py as detailed in HOWTO_RELEASE::
+
+    $ gvim pavement.py setup.py
+    $ git commit -a -m"REL: NumPy 1.12.0 release."
+
+Sanity check::
+
+    $ python runtests.py -m "full"
+    $ python3 runtests.py -m "full"
+
+Tag it,and build the source distribution archives::
+
+    $ git tag -s v1.12.0
+    $ paver sdist # sdist will do a git clean -xdf, so we omit that
+
+Check that the files in ``release/installers`` have the correct versions, then
+push the tag upstream; generation of the wheels for PyPI needs it::
+
+    $ git push upstream v1.12.0
+
+Trigger the wheels build. This can take a while. The numpy-wheels repository is
+cloned from `<https://github.com/MacPython/numpy-wheels>`_. Start with a pull
+as the repo may have been accessed and changed by someone else and a push will
+fail. The ``.travis.yml`` and ``appveyor.yml`` files need to be edited to make
+sure they have the correct version, search for ``BUILD_COMMIT``. If you get
+nervous at the amount of time taken -- the builds can take several hours-- you
+can check the build progress by following the links provided at
+`<https://github.com/MacPython/numpy-wheels>`_ to check the travis and appveyor
+build status.
+
+    $ cd ../numpy-wheels
+    $ git pull origin master
+    $ gvim .travis.yml appveyor.yml
+    $ git commit -a
+    $ git push origin HEAD
+
+Upload to PyPI
+--------------
+
+Upload to PyPI using the ``wheel-uploader``. The terryfy repository is
+cloned from `<https://github.com/MacPython/terryfy>`_. Uploading can be a
+hassle due to the site itself having problems. If you get strange messages, put
+it off for the next day. To begin, check if all the needed wheels have been
+built. There should be 22 of them at `<https://wheels.scipy.org>`_, 4 for Mac,
+8 for Windows, and 10 for Linux. Here the wheelhouse directory is in the home
+directory. Then upload the wheels. You will need to sign every file separately
+in the process, keeping the pass phrase in the clipboard and pasting it in will
+make that easier::
```

Comment:
I thought that also, but vaguely recall it not working. Probably need to test again. In any case, `wheel-uploader` doesn't do that and when I asked @matthew-brett  about it he said they could not get it to work.